### ISSUE TITLE
`to_yojson` expects safe_map to be directly in scope

### ISSUE THREAD

**Issue by MattWindsor91** [ocaml-ppx/ppx_deriving_yojson#100]
On `ppx_deriving_yojson` v3.5, OCaml 4.07.1, with both the ppx and the runtime library referenced in dune (and implicit transitive dependencies _off_):

```ocaml
let foo = [%to_yojson: int list] (* Unbound value safe_map *)

let safe_map = Ppx_deriving_yojson_runtime.safe_map
let bar = [%to_yojson: int list] (* ok *)
```

Nothing else seems to expect `Ppx_deriving_yojson_runtime` to be `open`, so I'm assuming this is a bug.

I haven't yet made a minimum working example project, but this seems to trigger even if the offending line is the first non-comment line in a module (ie, nothing else has been opened).

**Comment by gasche** [comment]
This seems to be a bug indeed... I saw some of these failures in the CI testing, but only on old versions of the affected software, so I didn't connect the dots. Silly me! I will try to get a fix ready at some point this week.

**Comment by gasche** [comment]
So it turns out that our testsuite does test the `list` deriver, but not the `[%of_json ..]`, `[%to_json ..]` forms where the problem occurs -- when using `[@@deriving yojson]` on the same type, the generation is currently wrapped with `Ppx_deriving_yojson_runtime` quantifications.

**Comment by gasche** [comment]
The bug was fixed in #101, and I'm going to release 3.5.1 with the fix.

**Comment by MattWindsor91** [comment]
Cheers!  3.5.1 appears to work fine now (except for the quick fix I made to bring `safe_map` into scope now triggering an unused warning :D)
